src/Common/src/Interop/Unix/libcurl/Interop.libcurl_multi.cs

```diff
@@ -46,16 +32,16 @@ internal static partial class libcurl
             SafeCurlHandle easy_handle);
 
         [DllImport(Interop.Libraries.LibCurl)]
-        public static extern int curl_multi_assign(
+        public static extern unsafe int curl_multi_wait(
             SafeCurlMultiHandle multi_handle,
-            curl_socket_t sockfd,
-            IntPtr sockptr);
+            curl_waitfd* extra_fds,
+            uint extra_nfds,
+            int timeout_ms,
+            out int numfds);
 
         [DllImport(Interop.Libraries.LibCurl)]
-        public static extern int curl_multi_socket_action(
+        public static extern int curl_multi_perform(
```

Comment:
The reason for choosing curl_multi_socket_action over curl_multi_perform was the foll. from section labelled MULTI_SOCKET(http://curl.haxx.se/libcurl/c/libcurl-multi.html)

>curl_multi_socket_action function offers a way for applications to not only avoid being forced to use select(), but it also offers a much more high-performance API that will make a significant difference for applications using large numbers of simultaneous connections. 

Currently we have used a platform-agnostic poll with lower perf but the hope is that we can move to a platform-specific performant version (maybe epoll on Linux etc.)